pkg/nfd-gc/nfd-gc.go

```diff
@@ -57,10 +62,11 @@ type NfdGarbageCollector interface {
 }
 
 type nfdGarbageCollector struct {
-	args     *Args
-	stopChan chan struct{}
-	client   metadataclient.Interface
-	factory  metadatainformer.SharedInformerFactory
+	args      *Args
+	stopChan  chan struct{}
+	client    metadataclient.Interface
+	factory   metadatainformer.SharedInformerFactory
+	k8sClient k8sclient.Interface
```

Comment:
We shouldn't need a separate client but just reuse the metadataclient above (we're not interested in the content of the podspec)

```suggestion
```